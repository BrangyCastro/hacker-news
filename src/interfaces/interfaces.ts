export interface News {
  story_id: number;
  story_title: string;
  created_at: string;
  author: string;
  story_url: string | null;
}

export interface Options {
  label: string;
  value: string;
  icon: string;
}

export interface TabsOptions {
  title: string;
  component: React.ReactElement;
}
