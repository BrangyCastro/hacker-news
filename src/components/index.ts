export { Header } from "./Header/Header";
export { Tabs } from "./Tabs/Tabs";
export { Card } from "./Card/Card";
export { Dropdown } from "./Dropdown/Dropdown";
