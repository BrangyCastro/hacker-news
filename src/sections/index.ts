export { Alls } from "./Alls/Alls";
export { Favorite } from "./Favorite/Favorite";
