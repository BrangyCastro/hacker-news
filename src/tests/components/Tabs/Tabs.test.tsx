import React from "react";
import { mount } from "enzyme";
import { Tabs } from "../../../components";
import { TabsOptions } from "../../../interfaces/interfaces";
import { Alls, Favorite } from "../../../sections";

describe("Tabs tests", () => {
  const tabsOptions: TabsOptions[] = [
    {
      title: "Alls",
      component: <Alls />,
    },
    {
      title: "My Faves",
      component: <Favorite />,
    },
  ];

  const wrapper = mount(<Tabs tabs={tabsOptions} />);

  test("should be displayed correctly", () => {
    expect(wrapper).toMatchSnapshot();
    const selectedDefault = wrapper.find(".selected").text().trim();
    expect(selectedDefault).toBe("Alls");
  });

  test("should be two tabs", () => {
    const tabs = wrapper.find("li");
    expect(tabs).toHaveLength(tabsOptions.length);
  });

  test("should select the My Faves tabs", () => {
    wrapper.find("li").at(1).simulate("click");
    const tabs = wrapper.find("Favorite").exists();
    expect(tabs).toBe(true);
    const tabs1 = wrapper.find("Alls").exists();
    expect(tabs1).toBe(false);
  });
});
