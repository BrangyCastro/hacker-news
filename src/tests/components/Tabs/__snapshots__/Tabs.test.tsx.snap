// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs tests should be displayed correctly 1`] = `
<div>
  <ul
    className="inline"
  >
    <li
      className="selected"
      key="0"
      onClick={[Function]}
    >
      Alls
    </li>
    <li
      className=""
      key="1"
      onClick={[Function]}
    >
      My Faves
    </li>
  </ul>
  <div
    className="tab"
  >
    Array [
      <div
        className="selector"
      >
        <div
          id="selectField"
          onClick={[Function]}
        >
          <p>
            Select your news
          </p>
          <img
            alt=""
            className=""
            src="arrow.png"
          />
        </div>
      </div>,
      <h1
        className="loading"
      >
        Loading news...
      </h1>,
    ]
  </div>
</div>
`;
